pkg/core/native/native_test/neo_test.go

```diff
@@ -0,0 +1,394 @@
+		e.CheckHalt(t, tx.Hash(), stackitem.Make(true)) // luckily, both `transfer`, `registerCandidate` and `vote` return boolean values
+	}
+	//transferBlock := e.Chain.BlockHeight()
+
+	// We still haven't voted enough validators in.
+	pubs, err = e.Chain.GetValidators()
+	require.NoError(t, err)
+	require.Equal(t, standBySorted, keys.PublicKeys(pubs))
+
+	advanceChain(t)
+	pubs, err = e.Chain.GetNextBlockValidators()
+	require.NoError(t, err)
+	require.EqualValues(t, standBySorted, keys.PublicKeys(pubs))
+
+	// Register and give some value to the last validator.
+	txes = txes[:0]
+	registerTx := neoValidatorsInvoker.WithSigners(candidates[0]).PrepareInvoke(t, "registerCandidate", candidates[0].(neotest.SingleSigner).Account().PrivateKey().PublicKey().Bytes())
+	txes = append(txes, registerTx)
+	voteTx := neoValidatorsInvoker.WithSigners(voters[0]).PrepareInvoke(t, "vote", voters[0].(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), candidates[0].(neotest.SingleSigner).Account().PrivateKey().PublicKey().Bytes())
+	txes = append(txes, voteTx)
+	neoValidatorsInvoker.AddNewBlock(t, txes...)
+	for _, tx := range txes {
+		e.CheckHalt(t, tx.Hash(), stackitem.Make(true)) // luckily, both `transfer`, `registerCandidate` and `vote` return boolean values
+	}
+
+	advanceChain(t)
+	pubs, err = neoCommitteeInvoker.Chain.GetNextBlockValidators()
+	require.NoError(t, err)
+	sortedCandidates := make(keys.PublicKeys, validatorsCount)
+	for i := range candidates[:validatorsCount] {
+		sortedCandidates[i] = candidates[i].(neotest.SingleSigner).Account().PrivateKey().PublicKey()
+	}
+	sort.Sort(sortedCandidates)
+	require.EqualValues(t, sortedCandidates, keys.PublicKeys(pubs))
+
+	pubs, err = neoCommitteeInvoker.Chain.GetNextBlockValidators()
+	require.NoError(t, err)
+	require.EqualValues(t, sortedCandidates, pubs)
+
+	t.Run("check voter rewards", func(t *testing.T) {
+		gasBalance := make([]*big.Int, len(voters))
+		neoBalance := make([]*big.Int, len(voters))
+		txes = make([]*transaction.Transaction, 0, len(voters))
+		for i := range voters {
+			h := voters[i].ScriptHash()
+			gasBalance[i] = e.Chain.GetUtilityTokenBalance(h)
+			neoBalance[i], _ = e.Chain.GetGoverningTokenBalance(h)
+			tx := neoCommitteeInvoker.WithSigners(voters[i]).PrepareInvoke(t, "transfer", h.BytesBE(), h.BytesBE(), int64(1), nil)
+			txes = append(txes, tx)
+		}
+		neoCommitteeInvoker.AddNewBlock(t, txes...)
+		for _, tx := range txes {
+			e.CheckHalt(t, tx.Hash(), stackitem.Make(true))
+		}
+
+		// GAS increase consists of 2 parts: NEO holding + voting for committee nodes.
+		// Here we check that 2-nd part exists and is proportional to the amount of NEO given.
+		for i := range voters {
+			newGAS := e.Chain.GetUtilityTokenBalance(voters[i].ScriptHash())
+			newGAS.Sub(newGAS, gasBalance[i])
+			// TODO: deal with unexported CalculateNEOHolderReward
+			gasForHold, err := e.Chain.CalculateNEOHolderReward(voters[i].ScriptHash(), e.Chain.BlockHeight())
```

Comment:
@roman-khimov, I'm not sure how to deal with this part of test. Here we need to compare voter rewards (second part), but `CalculateNEOHolderReward` is unexported.

Reply:
1. Check the value against non-voter (holder) account that holds NEO for the same number of blocks.
2. Hardcode exact GAS values generated for the tests.

Reply:
I'm not sure that hardcoded value is good and maintainable, so let's go with the first suggestion. I added these reference accounts and comment on why do we need them. Fixed.

Reply:
Well, it's a test, it can expect some exact value given known conditions, but comparing to non-voter is more flexible, of course.